### PR TITLE
Add a timestamp in  the block header

### PR DESCRIPTION
**Pull request by TrustHenry**
Synchronize Agora changes

https://github.com/bpfkorea/agora/pull/1437
Related to
https://github.com/bpfkorea/stoa/issues/241

### CODE REVIEW

**Comment by TrustHenry** [comment]
ping

**Comment by MichaelKim20** [comment]
The following code needs to be modified:
``` DIFF

--- a/src/modules/utils/JSONValidator.ts
+++ b/src/modules/utils/JSONValidator.ts
@@ -102,6 +102,9 @@ export class JSONValidator
                         },
                         "type": "array"
                     },
+                    "timestamp": {
+                        "type": "string"
+                    }
                 },
                 "additionalProperties": false,
                 "required": [
@@ -110,7 +113,8 @@ export class JSONValidator
                     "merkle_root",
                     "validators",
                     "signature",
-                    "enrollments"
+                    "enrollments",
+                    "timestamp"
                 ]
             }
         ],

```

**Comment by TrustHenry** [comment]
@MichaelKim20 Thanks for the review.  I apply the suggestion.

**Review by MichaelKim20 — APPROVED** [review]
It's very good for now. But I need to check if it is applied to Stoa and works properly so that it can be finally approved.

File: src/modules/utils/JSONValidator.ts [hunk 1]
```diff
@@ -102,6 +102,9 @@ export class JSONValidator
                         },
                         "type": "array"
                     },
+                    "timestamp": {
+                        "type": "string"
```
Comment (MichaelKim20) [inline src/modules/utils/JSONValidator.ts:106]:
```suggestion
                        "type": "number"
```

**Review by MichaelKim20 — CHANGES_REQUESTED** [review]

File: src/modules/utils/JSONValidator.ts (hunk 1 above)
Comment (MichaelKim20) [reply src/modules/utils/JSONValidator.ts:106]: {
  "prev_block": "0x8c1767e1c770ec82d597a01197eaccdcd04ff576a70c44beb73d14d98f9213e9b7c961ae22ce53c2dec51371f3331d847309280414b3a7967ebb217fe9fe94b3",
  "height": "9",
  "merkle_root": "0x6a0804c2f3cb3b0d64cd8cf2da34623ff43a0ccee1bd4be686d1ea3f3c3724f04e620acc0c2f2f02813c316607a81ac9963181b639e4b13cd53c04fcc2e18fbc",
  "validators": "[252]",
  "signature": "0x08842aff36b64eb13043f33416b07aad510bed53088024743746544061a646a540d334d2cdb3d745bd2ff2aedf6e10422bd6a8445e7df60d1ddc6158792379fc",
  "enrollments": [],
  "random_seed": "0x60d0e8ff49edc4c24756519dc2ed50c8195ec1847d415de413471bde04c93c3f268c071c22fd9b2dc7baae8c28c45c6bdd5fa74115e9fd6a1be37a324fa9e53f",
  "missing_validators": [],
  "timestamp": 1610418348
}

Comment (MichaelKim20) [reply src/modules/utils/JSONValidator.ts:106]:
``` TypeScript
        let a = new Date(4503599627370496);
        console.log(a.toDateString());
        let b = new Date(1610418348);
        console.log(b.toDateString());
```
Result : 
Thu May 24 144683
Tue Jan 20 1970

4503599627370496 is 2^52 (Number's Max)

So we can treat this as a Nuber.

**Review by MichaelKim20 — APPROVED** [review]

**Pull request merged**